### PR TITLE
Np 44714 has create rights

### PR DESCRIPTION
**Pull request by StigNorland**

### CODE REVIEW

File: publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java [hunk 1]
```diff
@@ -1,11 +1,13 @@
 package no.unit.nva.publication.create;
 
+import static nva.commons.apigateway.AccessRight.PUBLISH_THESIS;
```
Comment (ketilaa) [inline publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java:3]: PUBLISH_THESIS is bad naming imho. It is not about publishing, but rather creating a publication of certain types, isn't it?

Comment (StigNorland) [reply publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java:3]: I tried to keep it is style with the rest of AccessRights
```
    PUBLISH_FILES,
    PUBLISH_METADATA,
    PUBLISH_THESIS,
    PUBLISH_THESIS_READ_EMBARGOED_FILES,
    APPROVE_DOI_REQUEST,
    REJECT_DOI_REQUEST,
    READ_DOI_REQUEST,
    APPROVE_PUBLISH_REQUEST,
    EDIT_OWN_INSTITUTION_RESOURCES,
    EDIT_OWN_INSTITUTION_USERS,
    EDIT_OWN_INSTITUTION_PROJECTS,
    EDIT_OWN_INSTITUTION_PUBLICATION_WORKFLOW,
```
Feel free to come up with suggestions.

File: publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java [hunk 2]
```diff
@@ -99,21 +104,33 @@ private UserInstance createUserInstanceForInternalUser(RequestInfo requestInfo)
...
+    private boolean isThesisAndHasNotPublishThesis(RequestInfo requestInfo, CreatePublicationRequest request) {
+        var thesisKinds = List.of("DegreeBachelor", "DegreeMaster", "DegreePhd");
+        var kind =
+            attempt(() -> request.getEntityDescription().getReference().getPublicationInstance().getInstanceType())
+                .toOptional();
+
+        if (kind.isPresent() && thesisKinds.contains(kind.get())) {
+            return !requestInfo.userIsAuthorized(PUBLISH_THESIS.name());
+        }
+        return false;
```
Comment (truhacevkir) [inline publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java:134]: Maybe rename kind to publicationInstanceType and thesisKinds to static list THESES?


        if (isThesis(publicationInstanceType)) {
            hasRightsToPublicateThesis(requestInfo);
        }

File: publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationRequest.java [hunk 3]
```diff
@@ -31,21 +35,29 @@ public class UpdatePublicationRequest implements WithIdentifier, WithMetadata, W
     private List<Funding> fundings;
     private String rightsHolder;
     
-    public Publication generatePublicationUpdate(Publication existingPublication) {
+    public Publication generatePublicationUpdate(Publication existingPublication) throws ForbiddenException {
         if (!this.identifier.equals(existingPublication.getIdentifier())) {
             throw new IllegalArgumentException(
                 WRONG_PUBLICATION_UDPATE_ERROR + existingPublication.getIdentifier());
```
Comment (truhacevkir) [inline publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationRequest.java:41]: Typo

File: publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java [hunk 4]
```diff
@@ -65,27 +68,29 @@ public CreatePublicationHandler(ResourceService publicationService,
...
     }
-    
+
     @Override
     protected PublicationResponse processInput(CreatePublicationRequest input, RequestInfo requestInfo,
                                                Context context) throws ApiGatewayException {
         logger.info(attempt(() -> JsonUtils.dtoObjectMapper.writeValueAsString(requestInfo)).orElseThrow());
-        
-        var userInstance = createUserInstanceFromLoginInformation(requestInfo);
+        if (isThesisAndHasNotPublishThesis(requestInfo, input)) {
```
Comment (truhacevkir) [inline publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java:76]: *AndHasNotPublishThesis is not the best method name. You can do something like 

hasRightsToPublishThesis(input, requestInfo);

or see comment below

File: publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java (hunk 1 above)
Comment (StigNorland) [reply publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java:3]: EDIT_OWN_INSTITUTION_RESOURCE_THESIS ?

Comment (StigNorland) [reply publication-rest/src/main/java/no/unit/nva/publication/create/CreatePublicationHandler.java:3]: Are there users that has access to other institutions resources?

**Review by ketilaa — APPROVED** [review]

**Pull request merged**